x-pack/platform/packages/shared/onechat/onechat-common/tools/tool_result.ts

```diff
@@ -66,3 +69,8 @@ export type ToolResult =
   | QueryResult
   | OtherResult
   | ErrorResult;
+
+const charset = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz';
+export function getToolResultId(len = 4): string {
+  return Array.from({ length: len }, () => charset[randomInt(charset.length)]).join('');
+}
```

Comment:
Move that either in `@kbn/onechat-server`, `@kbn/onechat-genai-utils` or even closer to where it's being used in the onechat plugin's server-side code.